### PR TITLE
Feature/refactoring

### LINKED ISSUE #1: Cannot get default office
after creating a default office (without any alias), i cannot get it using
```ts
enmail.getOffice()
```
because method `Expected 1 arguments, but got 0.`.

I think as it is optional at the time of the creation, it should be optional here with same default values.

### LINKED ISSUE #2: Cannot import Mail interface directly
In a typescript project, Mail can be only imported from dist/Mail and not directly from the package

✔️ 
```ts
import { Mail } from 'enmail/dist/Mail';
```

❌
```ts
import { Mail } from 'enmail';
```

I think the other one should be the only one working.

### LINKED ISSUE #3: Incorrect types for `compileLodashTemplate()`
returns `any`, should be a function with specific context options returning a `string`.

### LINKED ISSUE #4: Translation variable vs docs mismatch
In the docs (readme) a double underscore variable `__` is described as a _common value for translations_, but the implementation injects only a single underscore (`_`) for this purpose.

### PR DESCRIPTION
**Pull request by stefan-prokop-cz**
This closes #1, closes #2, closes #3, closes #4

### CODE REVIEW

File: src/offices/SendgridOffice.ts [hunk 1]
```diff
@@ -17,7 +16,7 @@ export class SendgridOffice extends Office {
     public mailer!: SendgridMailer;
     constructor(options: SendgridOfficeOptions) {
         sendgrid.setApiKey(options.apiKey);
-        super(sendgrid);
+        super(sendgrid as unknown as SendgridMailer);
```
Comment (vlasy) [inline src/offices/SendgridOffice.ts]: 👍 `unknown` jsem ani neznal :)

File: src/test/utils/testMock.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* tslint:disable:no-console */
```
Comment (vlasy) [inline src/test/utils/testMock.ts]: I prefer 
```typescript
const logger = console;
...
logger.log(...)
```

File: src/offices/SendgridOffice.ts (hunk 1 above)
Comment (vlasy) [inline src/offices/SendgridOffice.ts]: 👍

File: src/templates/withDefaultContext.ts [hunk 3]
```diff
@@ -1,9 +1,6 @@
+import intl = require('intl');
```
Comment (smoliji) [inline src/templates/withDefaultContext.ts]: ❗️ import + require. What is this sorcery?! :O

Comment (smoliji) [reply src/templates/withDefaultContext.ts]: Ive stumbled upon it only here https://palantir.github.io/tslint/rules/no-var-requires/, but cannot find any official spec for this

Comment (smoliji) [reply src/templates/withDefaultContext.ts]: Found it! Its a Typescript specific feature https://www.typescriptlang.org/docs/handbook/modules.html. Section export = and import = require(). 💡

File: src/test/utils/testMock.ts (hunk 2 above)
Comment (stefan-prokop-cz) [reply src/test/utils/testMock.ts]: Changed

File: src/index.ts [hunk 4]
```diff
@@ -1,20 +1,21 @@
...
+export * from './createOffice';
+export * from './Mail';
+export * from './offices/FcmOffice';
+export * from './offices/GmailOffice';
+export * from './offices/OnesignalOffice';
+export * from './offices/Office';
+export * from './offices/SendgridOffice';
+export * from './templates/compileLodashFileTemplate';
+export * from './templates/compileLodashTemplate';
+export * from './templates/withDefaultContext';
```
Comment (grissius) [inline src/index.ts]: How are globs better than the explicit exports? The index should only export explicitly IMO, so you cannot change the public API by accident.

File: src/offices/GmailOffice.ts [hunk 5]
```diff
@@ -32,7 +32,7 @@ export class GmailOffice extends Office {
         const definedMailer = options.mailer || nodemailer;
         const mailer = definedMailer.createTransport({
             service: ServiceType.gmail,
-
+             // tslint:disable-next-line
```
Comment (grissius) [inline src/offices/GmailOffice.ts]:
```suggestion
```
It does not seem to be necessary

File: src/templates/withDefaultContext.ts (hunk 3 above)
Comment (grissius) [reply src/templates/withDefaultContext.ts]:
```suggestion
import * as intl from 'intl';
```
This is not classy :tophat: and unaccaptable. It is excused only when usign `export = ` sytnax for class/interface/type to make TS happy, as stated in the docs.

File: src/templates/withDefaultContext.ts [hunk 6]
```diff
@@ -1,9 +1,6 @@
...
 import { defaults } from 'lodash';
-
-
-const intl = require('intl');
-const moment = require('moment');
-require('moment-timezone');
+import moment = require('moment');
```
Comment (grissius) [inline src/templates/withDefaultContext.ts]:
```suggestion
import * as moment from 'moment';
```
Not classy :cocktail:

File: src/templates/withDefaultContext.ts [hunk 7]
```diff
@@ -12,7 +9,7 @@ export const withDefaultContext = (context: any = {}) => {
 
     const defaultContext: object = {
         // common var for translation
-        _: ((...args: any[]) => (context.__ ? context.__(...args) : String(args))),
+        __: ((...args: any[]) => (context.__ ? context.__(...args) : String(args))),
```
Comment (grissius) [inline src/templates/withDefaultContext.ts]: Why suddenly two? I18n convention or other reason?

File: src/templates/withDefaultContext.ts [hunk 8]
```diff
@@ -12,7 +9,7 @@ export const withDefaultContext = (context: any = {}) => {
 
     const defaultContext: object = {
```
Comment (grissius) [inline src/templates/withDefaultContext.ts]: Understand it is not this PR change, but typehint here actually destroys the type and casts to generic object.

File: src/templates/withDefaultContext.ts (hunk 7 above)
Comment (stefan-prokop-cz) [reply src/templates/withDefaultContext.ts]: It was in previous version and i18n it also uses

File: src/templates/withDefaultContext.ts (hunk 8 above)
Comment (stefan-prokop-cz) [reply src/templates/withDefaultContext.ts]: Removed

File: src/index.ts (hunk 4 above)
Comment (stefan-prokop-cz) [reply src/index.ts]: Replaced with explicit exports

**Comment by smoliji** [comment]
One more thing @stefan-prokop-cz , add to the PR description, which issues this fixes. See [how to](https://help.github.com/en/articles/closing-issues-using-keywords#closing-multiple-issues) for more details.

**Comment by stefan-prokop-cz** [comment]
This closes #1, closes #2, closes #3, closes #4

**Pull request merged**